src/praisonai-agents/tests/test_plugin_system.py

```
@@ -559,32 +559,44 @@ def run(self, a: float, b: float) -> float:
         result = agent.execute_tool("power", {"base": 2, "exp": 3})
         assert result == 8
-        
+
+        # Default agent (no opt-in) must NOT reach the global registry.
+        safe_agent = Agent(
+            name="SafeAgent",
+            role="Tester",
+            goal="Test isolation",
+            tools=[],
+        )
+        with pytest.raises(Exception):
```

Ruff (0.15.21)

[warning] Do not assert blind exception: `Exception`

(B017)